openedx/core/djangoapps/course_roles/models.py

```diff
@@ -0,0 +1,86 @@
+"""
+Models for course roles schema
+"""
+from django.contrib.auth import get_user_model
+from django.db import models
+
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from organizations.models import Organization
+
+User = get_user_model()
+
+
+class CourseRolesRole(models.Model):
+    """
+    Model for a course role.
+    """
+    class RoleType(models.TextChoices):
+        """
+        Enum for the role type.
+        """
+        DEFAULT = 'default'
+        CUSTOM = 'custom'
+    name = models.CharField(max_length=255)
+    short_description = models.CharField(max_length=255)
+    long_description = models.TextField()
+    type = models.CharField(max_length=64, choices=RoleType.choices, default=RoleType.DEFAULT)
+    service = models.ForeignKey('CourseRolesService', on_delete=models.DO_NOTHING, null=True)
+    permissions = models.ManyToManyField('CourseRolesPermission', through='CourseRolesRolePermissions')
+    users = models.ManyToManyField(User, through='CourseRolesUserRole')
+
+    def __str__(self):
+        return self.name
+
+
+class CourseRolesPermission(models.Model):
+    """
+    Model for a course roles permission.
+    """
+    name = models.CharField(max_length=255, unique=True)
+    description = models.CharField(max_length=255)
+
+    def __str__(self):
+        return self.name
+
+
+class CourseRolesRolePermissions(models.Model):
+    """
+    Model for a course roles role permission.
+    """
+    role = models.ForeignKey('CourseRolesRole', on_delete=models.CASCADE)
+    permission = models.ForeignKey('CourseRolesPermission', on_delete=models.CASCADE)
+    allowed = models.BooleanField(default=False)
```

Comment:
Is the intent that having a course role permission with `allowed` set to False amounts to saying that that role on that course lacks the permission? How does this compare with having permissions that exist but are not listed on the CourseRolesRolePermissions object? If they're the same, what's the benefit of having an "allowed" field? Why not just list only those permissions in CourseRolesRolePermissions for which `allowed` is True?

Reply:
This field is also related to custom roles, I will remove it for the moment.